assayist/common/models/content.py

```diff
@@ -50,7 +50,29 @@ class Artifact(AssayistStructuredNode):
     build = RelationshipFrom('Build', 'PRODUCED', cardinality=ZeroOrOne)
     # The artifacts that were in the buildroot when this artifact was built
     buildroot_artifacts = RelationshipTo('Artifact', 'BUILT_WITH')
+    # The external artifacts that were in the buildroot when this artifact was built
+    buildroot_external_artifacts = RelationshipTo('ExternalArtifact', 'BUILT_WITH')
     # The artifacts that are embedded in this artifact
     embedded_artifacts = RelationshipTo('Artifact', 'EMBEDS')
+    # The external artifacts that are embedded in this artifact
+    embedded_external_artifacts = RelationshipTo('ExternalArtifact', 'EMBEDS')
     # The source locations that embedded in this artifact
     embedded_source_locations = RelationshipTo('.source.SourceLocation', 'EMBEDS')
+
+
+class ExternalArtifact(AssayistStructuredNode):
+    """
+    The definition of an ExternalArtifact node.
+
+    These represent artifacts that are not built internally and are downloaded directly from
+    upstream and used during builds.
+    """
+
+    checksum = StringProperty()
+    identifier = StringProperty(unique_index=True)
```

Comment:
It's not clear to me right now what one of these would look like.

Reply:
This would be type-specific. Something like `org.apache.maven.plugins:maven-compiler-plugin:jar:3.1` to signify that this version of the java compiler plugin was used.